runtime/runtime/tests/test_async_calls.rs

```diff
@@ -371,63 +358,68 @@ fn test_two_promises_with_two_callbacks() {
     let [r0] = &*assert_receipts!(group, signed_transaction) else {
         panic!("Incorrect number of produced receipts")
     };
-    let [r1, cb1, ref0] = &*assert_receipts!(group, "near_0" => r0 @ "near_1",
-    ReceiptEnum::Action(ActionReceipt{actions, ..}), {},
-    actions,
-    a0, Action::FunctionCall(function_call_action), {
-       assert_eq!(function_call_action.gas, GAS_1);
-       assert_eq!(function_call_action.deposit, 0);
+
+    let receipts = &*assert_receipts!(group, "near_0" => r0 @ "near_1",
+        ReceiptEnum::Action(ActionReceipt{actions, ..}), {},
+        actions,
+        a0, Action::FunctionCall(function_call_action), {
+            assert_eq!(function_call_action.gas, GAS_1);
+            assert_eq!(function_call_action.deposit, 0);
+        }
+    );
+    let [r1, cb1] = &receipts[..2] else { panic!("Incorrect number of produced receipts") };
+    if ProtocolFeature::ReducedGasRefunds.enabled(PROTOCOL_VERSION) {
+        assert_eq!(receipts[2..], [], "refund should have been avoided");
+    } else {
+        let [refund] = &receipts[2..] else { panic!("Incorrect number of refunds") };
+        assert_refund!(group, refund @ "near_0");
     }
```

Comment:
As a follow-up to the comment above you can do

```suggestion
    let [r1, cb1, refunds@..] = &receipts;
    if ProtocolFeature::ReducedGasRefunds.enabled(PROTOCOL_VERSION) {
        assert!(refunds.is_empty(), "refund should have been avoided");
    } else {
        let [refund] = refunds else { panic!("Incorrect number of refunds") };
        assert_refund!(group, refund @ "near_0");
    }
```

Reply:
Nice, I didn't know about this syntax! Yeah that makes some slice juggling much nicer. Still doesn't reduce the number of panics I need to write though, unfortunately.

Reply:
Oh and btw I prefer `assert_eq!(refunds, [])` over `assert!(refunds.is_empty()` because it will show me the produced receipts in the error messages.